examples/profiling/README.md

```diff
@@ -1,59 +1,91 @@
+In this example we demonstrate how to profile on the RISC Zero zkVM guest programs, using the Fibonacci sequence calculation as an example.
 
-## Overview
+<!-- NOTE: This content matches the docs under website/api/zkvm/developer-guide/profiling.md -->
+## Background
 
-There are three different Fibonacci sequence calculation methods provided:
+Profiling tools, like [pprof] and [perf], allow collecting performance information over the entire execution of your program, and help create visualizations for the performance of your program.
+RISC Zero has experimental support for generating pprof files for cycle counts.
 
-1. `fibonacci_1`: A basic iterative method.
-2. `fibonacci_2`: An optimized iterative method that attempts to batch computation.
-3. `fibonacci_3`: A matrix exponentiation approach, which is a fast method to compute Fibonacci numbers.
-
-The host code sets up a profiling environment and then runs the RISC0 guest code that computes Fibonacci numbers using the three methods. After execution, the profiling data is written out for further analysis.
+[Sampling CPU profilers], as implemented by pprof and perf, provide a view of where your program is spending its time.
+It does so by recording the current call stack at a sampling interval.
+RISC Zero provides a "sampling" [^1] CPU profiler for guest execution.
 
-The guest code reads the number of iterations from the host, computes the Fibonacci number, and finally commits the answer back to the host.
+[pprof]: https://github.com/google/pprof
+[perf]: https://perf.wiki.kernel.org/index.php/Main_Page
+[Sampling CPU profilers]: https://nikhilism.com/post/2018/sampling-profiler-internals-introduction/
 
 ## Usage
 
 ### Step 1: Prerequisites
+
 First, [install Rust] if you don't already have it.
+Additionally, you will need to [install Go], which bundles with it the [pprof] tool.
 
 Next, install the `cargo-risczero` tool and install the toolchain with:
+
 ```bash
 cargo install cargo-binstall
 cargo binstall cargo-risczero
 cargo risczero install
 ```
 
-Finally, install the [pprof] tool with
-```bash
-go install github.com/google/pprof@latest
-```
-
 ### Step 2: Running
-Then, run the example with:
+
+Run the Fibonacci profiling example with:
+
 ```bash
-RISC0_PPROF_OUT=./profile.pb cargo run --release
+RISC0_PPROF_OUT=./profile.pprof cargo run
 ```
 
-The above command will run the Fibonacci computation for 1000 iterations and write the profiling output to profile.pb.
+The above command will run the Fibonacci computation for 1000 iterations and write the profiling output to `profile.pprof`.
+Use the environment variable `RISC0_PPROF_OUT` to set to the desired output path for the profiling data.
 
 ### Step 3: Visualization
-To visualize the profiling data using `pprof`, run:
+
+To visualize the profile using `pprof`, run:
+
 ```bash
-pprof -http=127.0.0.1:8089 ../target/riscv-guest/riscv32im-risc0-zkvm-elf/release/fibonacci profile.pb
+go tool pprof -http=127.0.0.1:8000 profile.pprof
```

Comment:
This won't work unless someone has `go` installed.

I think we should avoid bringing in go, are there other installation methods that don't require go?

Reply:
This seems useful: https://github.com/tikv/pprof-rs

Reply:
I've been searching for a way to use pprof without installing Go, but can't find one that seems reasonable [1]. The method recommended in this doc before was to use `go install github.com/google/pprof@latest`, which then puts the `pprof` binary in `PATH`, but that is an extra step beyond simply installing Go, which already bundles pprof in.

[1] There exists a package _someone_ maintains called `gperftools` that includes it... but it's not widely used

Reply:
We could build and bundle it with `cargo-risczero` binstall

Reply:
I suppose this is fine given the context, maybe we should add a quick blurb about installing go.

Reply:
> This seems useful: https://github.com/tikv/pprof-rs

I don't think this crate can help us. It's a way to add profiling outputs to Rust programs by including the profiler in the binary. As it is designed, this would need to compiled into the guest binary, and would try to use Linux syscalls from within the guest to create a profile. So this tool won't generate a profile.

What we actually need is the `pprof` tool binary on the users machine. The tool is what does the visualization. It you look at `pprof-rs` docs, it actually points the user to use `pprof` installed via Go for visualization https://github.com/tikv/pprof-rs#use-with-pprof

Reply:
Oh, I just noticed (from the `pprof-rs` page) that the `gperftools` package is the old C++ implementation of pprof, which I don't think is being maintained anymore. IIUC, it also does not support the `http` server for interactively exploring profiles.